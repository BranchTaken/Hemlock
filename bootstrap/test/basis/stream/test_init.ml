open! Basis.Rudiments
open! Basis
open Stream

let test () =
  let rec test_init_up_to i n = begin
    match i <= n with
    | false -> ()
    | true -> begin
        let t = init (0L =:< i) ~f:(fun i -> i) in
        File.Fmt.stdout
        |> Fmt.fmt "init "
        |> Uns.pp i
        |> Fmt.fmt " ~f:(fun i -> i) = "
        |> (pp Uns.pp) t
        |> Fmt.fmt "\n"
        |> ignore;
        test_init_up_to (succ i) n
      end
  end in
  test_init_up_to 0L 3L

let _ = test ()
