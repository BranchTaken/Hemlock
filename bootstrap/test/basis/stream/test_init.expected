init 0 ~f:(fun i -> i) = Nil
init 1 ~f:(fun i -> i) = (0 Nil)
init 2 ~f:(fun i -> i) = (0 (1 Nil))
init 3 ~f:(fun i -> i) = (0 (1 (2 Nil)))
